src/item_factory.cpp

```diff
@@ -414,6 +414,18 @@ void Item_factory::finalize_post( itype &obj )
             }
         }
     }
+
+    if( !obj.ascii_picture.empty() ) {
+        std::vector<std::string> tmp_ascii_pic;
+        for( std::string line : obj.ascii_picture ) {
+            if( line.length() > 42 ) {
+                line = line.substr( 0, 42 );
+                debugmsg( "ascii_picture in %s contains a line too long to be displayed (>42 char).", obj.id );
+            }
+            tmp_ascii_pic.emplace_back( line );
```

Comment:
`tmp_ascii_pic` is excessive here, you can operate on `ascii_picture` directly if you don't plan to extract a named function.